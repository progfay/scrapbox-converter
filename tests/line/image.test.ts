/* global describe it expect */

import { parse } from '@progfay/scrapbox-parser'
import convert from '../../src'
import pretty from 'pretty'

describe('image', () => {
  it('Simple image', () => {
    const input = `Image
[https://i.gyazo.com/da78df293f9e83a74b5402411e2f2e01.png]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('Image with link', () => {
    const input = `Image
[http://cutedog.com https://i.gyazo.com/da78df293f9e83a74b5402411e2f2e01.png]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('Simple gyazo', () => {
    const input = `Image
[https://gyazo.com/da78df293f9e83a74b5402411e2f2e01]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('Gyazo with link', () => {
    const input = `Image
[http://cutedog.com https://gyazo.com/da78df293f9e83a74b5402411e2f2e01]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })
})
