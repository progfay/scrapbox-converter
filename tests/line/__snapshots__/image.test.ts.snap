// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image Gyazo with link 1`] = `
"<article>
  <h1 class=\\"title\\">Image</h1>
  <section>
    <p>
      <amp-img src=\\"https://gyazo.com/da78df293f9e83a74b5402411e2f2e01/thumb/1000\\" layout=\\"intrinsic\\" height=\\"360\\" width=\\"640\\" alt=\\"image\\" class=\\"normal-image\\"></amp-img>
    </p>
  </section>
</article>"
`;

exports[`image Image with link 1`] = `
"<article>
  <h1 class=\\"title\\">Image</h1>
  <section>
    <p>
      <amp-img src=\\"https://i.gyazo.com/da78df293f9e83a74b5402411e2f2e01.png\\" layout=\\"intrinsic\\" height=\\"360\\" width=\\"640\\" alt=\\"image\\" class=\\"normal-image\\"></amp-img>
    </p>
  </section>
</article>"
`;

exports[`image Simple gyazo 1`] = `
"<article>
  <h1 class=\\"title\\">Image</h1>
  <section>
    <p>
      <amp-img src=\\"https://gyazo.com/da78df293f9e83a74b5402411e2f2e01/thumb/1000\\" layout=\\"intrinsic\\" height=\\"360\\" width=\\"640\\" alt=\\"image\\" class=\\"normal-image\\"></amp-img>
    </p>
  </section>
</article>"
`;

exports[`image Simple image 1`] = `
"<article>
  <h1 class=\\"title\\">Image</h1>
  <section>
    <p>
      <amp-img src=\\"https://i.gyazo.com/da78df293f9e83a74b5402411e2f2e01.png\\" layout=\\"intrinsic\\" height=\\"360\\" width=\\"640\\" alt=\\"image\\" class=\\"normal-image\\"></amp-img>
    </p>
  </section>
</article>"
`;
